src/bsp/src/stm32/include/HostUart.h

```diff
@@ -0,0 +1,49 @@
+#ifndef __HOSTUART_H__
+#define __HOSTUART_H__
+
+#include "bsp/ICRC.h"
+#include "bsp/IHostUart.h"
+#include <FreeRTOS.h>
+#include <cstdint>
+#include <logger/ILogger.h>
+#include <semphr.h>
+
+class HostUart : public IHostUart {
+  public:
+    explicit HostUart(ICRC& crc, ILogger& logger);
+    ~HostUart() override = default;
+
+    bool send(const uint8_t* buffer, uint16_t length) override;
+    bool isBusy() override;
+
+    void process();
+
+    friend void hostUart_C_txCpltCallback(void* hostUartInstance);
+    friend void hostUart_C_rxCpltCallback(void* hostUartInstance);
+
+  private:
+    enum RxState { Rx_Idle, Rx_WaitForHeader, Rx_WaitForPayload, Rx_CheckIntegrity };
+    enum TxState { Tx_Idle, Tx_SendHeader, Tx_SendPayload };
+
+    ICRC& m_crc;
+    ILogger& m_logger;
+
+    TxState m_txState;
+    uint8_t m_txLength;
+    uint8_t m_txHeader[HOST_UART_HEADER_LENGTH];
+    const uint8_t* m_txBuffer;
+
+    uint8_t m_rxBuffer[HOST_UART_MAX_MESSAGE_LENGTH];
+    uint8_t m_rxHeader[HOST_UART_HEADER_LENGTH];
```

Comment:
std::array, easier to get the length after instead of using the defines (and avoiding the mistake of using the wrong define)